### DISCUSSION TITLE
RAVE name

### DISCUSSION THREAD

**Discussion by joewheaton** [Riverscapes/QRAVEPlugin#83] [Ideas] [answered]
I'm wondering if we should change the name of RAVE? The origin of RAVE as a name is from a drive on the North Island and New Zealand and the adolescent tendencies of @joewheaton and @philipbaileynar to make up stupid names. We came up with a stupid acronym first based on our habitat and obsession with having vowels in our names (e.g. BR**A**T, RC**A**T, so it ought to be R**A**VE) and that was _cute_ because you know, a RAVE. Well, a few things bug me:
- RAVE suggests a big party instead of anything useful here. 
- The important letters here are really R and V for riverscapes and visualization. However, we don't do anything fancy with visualization, its really basic. 
- **A** really bugs me as it is ambiguous whether we imply the noun *analysis* or hint at the verb *to analyze*. Either way, RAVE IS NOT where you do analysis. It is where you view a previous analysis. 
- **R** is everything - riverscapes.
- **E** explorer is a little redundant with GIS and with what the Warehouse is really becoming (i.e. a place explore and discover projects). 

We will need a logo to #82!

## The Suggestion

Maybe **RAVE** changes from the mouthful - "Riverscapes Analysis Visualization Explorer" to **RV** or the "Riverscapes Viewer". Quite simply, this is the place you view Riverscapes Project.  Plus, there are some fun icons we could use: 
![image](https://user-images.githubusercontent.com/9723905/196525831-c7724f82-ef82-4c57-b398-facdbf361420.png)

We will obviously need Aida's steer on imagery and  professional graphics designer. However, simplifying our suite of tools and acronyms and focusing on the main message is my motivation here. Sorry for the headaches this causes (e.g. redirects from http://rave.riverscapes.net to http://rv.riverscapes.net) if we adopt such a change.

I know that this is annoying for things like Repo Names (we have [QRAVEPlugin](https://github.com/Riverscapes/QRAVEPlugin) and then [RaveAddIn](https://github.com/Riverscapes/RaveAddIn) for the ArcGIS version, which I never really liked. And then I don't really know the various repos that @MattReimer uses to make WebRAVE do its magic.

But the Flavors would be:
- WebRV - Web Riverscapes Viewer
- ArcRV - ArcGIS Riverscapes Viewer (for 10.x) and ArcPro Riverscapes Viewer (for ArcPro soon)
- QRV - QGIS Riverscapes Viewer  

@philipbaileynar your thoughts? @wally-mac yours? @jtgilbert? @nick4rivers?

**Comment by joewheaton** [comment, marked as answer]
Okay, I've heard from various folks and so far everyone is fine with **RV** or actually enthusiastically likes the name change (and a few "meh... don't care"s). 

I'm going to consider this answered.  we will chaneg the name to **Riverscapes Viewer** or **RV**. Speak now or forever live out of an RV.

**Reply by MattReimer** [reply]
I'm not really opposed to a name change, except for all the links, strings and domain work that need to get done... which incentivizes us to do this the minimum number of times 

The only real concern I have is that "RV" isn't a very unique thing to search for and sometimes search engines ignore less than 3 characters. The QGIS plugin store seems to be fine but when I search for "rv" I get hundreds of results that we are going to have to compete with (because rv exists in the middle of words).

Searching for "Riverscapes" gives us just our tools though so maybe we tell people to search for "riverscapes" instead of just "RV"

Edit: QRV should be fine to search for.

**Reply by philipbaileynar** [reply]
I like QRV

**Reply by joewheaton** [reply]
Okay, so:
- For Repository Names and/or Abbreviations:
  - QRAVEPlugin → `QRV`
  - [RAVEAddIn](https://github.com/Riverscapes/RaveAddIn) → `ArcRV` (I'm assuming that when we do ArcPro we can do this from same repo like is done for Fluvial Coridor Toolbox)
  - WebRAVE → stays as part of https://github.com/Riverscapes/rs-web-monorepo and does not need renaming. When abbreviated, just `webRV`.  

We will refer to the whole collection as "Riverscapes Viewers" and "RV" for short.  

We will need to remap https://rave.riverscapes.net to  https://rv.riverscapes.net and handle the redirects. 

Should we keep the website hosted from the https://github.com/Riverscapes/RaveAddIn/Docs? It seems fine to me, just asking.

**Reply by MattReimer** [reply]
We need to figure out the scope of this change. We use the word RAVE in a lot of changes so we should choose our battles carefully.

What work do we need to do?

### RiverscapesXML

- We have a folder called `RaveBusinessLogic`. We can rename this to just `BusinessLogic` so we don't have to do this work again but this is the scariest change and affects all versions of RAVE code. I'd be tempted to not even change this.

### Code

- String changes: there are over 200 in this organization. We should audit and see which apply.
- Links in documentation and markdown

### OSGEO

- Need to rename the plugin on OSGEO (not sure this is possible but we can definitely publish it as a new plugin).

### Github

- Github tags - More just for completeness but easy and low risk so let's do it
- Repository names
  - https://github.com/Riverscapes/RaveAddIn. --> https://github.com/Riverscapes/ArcRV
  - https://github.com/Riverscapes/QRAVEPlugin --> https://github.com/Riverscapes/QRV

Repository renames are a little disruptive to developers but shouldn't affect anyone else as long as the new versions of the plugins change their references. Also I think github has a redirect feature we should look into

-------------

No ETA on this work yet but I predict this would be a smaller task than the `riverscapes.xyz` to `riverscapes.net` rename. Maybe only half as much work spread over 2-3 developers.

----------------

> Should we keep the website hosted from the https://github.com/Riverscapes/RaveAddIn/Docs? It seems fine to me, just asking

I think that's fine for now just to keep things simple

**Reply by joewheaton** [reply]
@MattReimer, you are the authority on answers to these good questions. I will just share my perspective below.

> We need to figure out the scope of this change. We use the word RAVE in a lot of changes so we should choose our battles carefully.

Agreed. I am okay with some things (e.g. Repo Names or Business Logic names still having the legacy RAVE name as necessary. 

> What work do we need to do?
> 
> ### RiverscapesXML
> * We have a folder called `RaveBusinessLogic`. We can rename this to just `BusinessLogic` so we don't have to do this work again but this is the scariest change and affects all versions of RAVE code. I'd be tempted to not even change this.

As I said above, I am okay with things people will not see not changing (e.g. stuff in the code and our repos). I am most concerned with stuff on public-facing websites, tool url, documentation and Tools itself. 

> ### Code
> * String changes: there are over 200 in this organization. We should audit and see which apply.
> * Links in documentation and markdown
> 
> ### OSGEO
> * Need to rename the plugin on OSGEO (not sure this is possible but we can definitely publish it as a new plugin).

We may want to talk about a strategy here anyway with @philipbaileynar as it relates to this and QRiS and when we finally go away from these experimental releases. 

> ### Github
> * Github tags - More just for completeness but easy and low risk so let's do it
> * Repository names
>   
>   * https://github.com/Riverscapes/RaveAddIn. --> https://github.com/Riverscapes/ArcRV
>   * https://github.com/Riverscapes/QRAVEPlugin --> https://github.com/Riverscapes/QRV
> 
> Repository renames are a little disruptive to developers but shouldn't affect anyone else as long as the new versions of the plugins change their references. Also I think github has a redirect feature we should look into

I like.

> No ETA on this work yet but I predict this would be a smaller task than the `riverscapes.xyz` to `riverscapes.net` rename. Maybe only half as much work spread over 2-3 developers.

Understood. 

> > Should we keep the website hosted from the https://github.com/Riverscapes/RaveAddIn/Docs? It seems fine to me, just asking
> 
> I think that's fine for now just to keep things simple

Well, @philipbaileynar any thoughts. I am mindful that the QRV is the repository we will most often be actually using and updating and pushing releases (i.e. with new documentation). It is a pretty simple thing to switch. If we are going to switch, now is the time. I agree it is not mandatory, but now is the time to speak up if we are uncomfortable.
